onnxruntime/python/tools/quantization/quantize.py

```diff
@@ -692,17 +693,26 @@ def quantize_dynamic(
 def quantize(
     model_input: Union[str, Path, onnx.ModelProto],
     model_output: Union[str, Path],
-    quant_config: QuantConfig,
+    quant_config: Union[QuantConfig, WeightOnlyQuantConfig],
 ):
     """Quantize a model with QuantConfig.
 
     Args:
         model_input (str | Path | ModelProto): Path to the model or ModelProto to quantize.
         model_output (str | Path): Path to save the quantized model.
-        quant_config (QuantConfig): Quantization Configuration.
+        quant_config (QuantConfig | WeightOnlyQuantConfig): Quantization Configuration.
     """
 
-    if isinstance(quant_config, StaticQuantConfig):
+    if isinstance(quant_config, WeightOnlyQuantConfig):
+        model = (
+            save_and_reload_model_with_shape_infer(model_input)
```

Comment:
>save_and_reload_model_with_shape_infer [](http://example.com/codeflow?start=12&length=38)

we don't need shape inference for weightOnly quantization because the shape of initializer is determined without shape inference.